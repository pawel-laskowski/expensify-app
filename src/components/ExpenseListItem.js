import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ( { description, amount, id, createdAt}) => (
  <div>
    <Link to={`edit/${id}`}>
      <h2>Description: {description}</h2>
    </Link>
    <p>amount: {amount}</p>  
  </div>
)

export default ExpenseListItem