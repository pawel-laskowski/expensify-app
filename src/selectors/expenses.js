import moment from "moment"

//get VisibleExpenses
export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.createdAt)
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day'): true
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day'): true
    const textMatch = typeof text !== 'string' || expense.description.toLocaleLowerCase().includes(text.toLocaleLowerCase())
    
    return startDateMatch && endDateMatch && textMatch
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1
    } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1 : -1
    }
  })
}