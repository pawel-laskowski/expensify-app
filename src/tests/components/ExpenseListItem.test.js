import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem"
import expenses from "../fixtures/expenses"

test('should render ExpenseListeItem correctly', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
  expect(wrapper).toMatchSnapshot()
})