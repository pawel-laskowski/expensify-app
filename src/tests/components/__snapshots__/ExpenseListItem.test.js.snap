// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListeItem correctly 1`] = `
<div>
  <Link
    to="edit/1"
  >
    <h2>
      Description: 
      Gum
    </h2>
  </Link>
  <p>
    $1.95
    -
    January 1st, 1970
  </p>
</div>
`;
